### DISCUSSION TITLE
Assistance with Using the http_get Function in Supabase

### DISCUSSION THREAD

**Discussion by fynn23** [supabase/supabase#19577] [Questions] [answered]
I am currently working on an SQL query where I am using the http_get function to retrieve data from the Stripe API. Unfortunately, I am having trouble writing the query correctly, and I need assistance to ensure that it works properly.

Here is the query I have written so far:

BEGIN

SELECT status, content::jsonb
FROM http_get(
  'https://api.stripe.com/v1/invoices',
  '{"customer": "cus_XXXXXX"}',
  'application/json',
  'headers' => jsonb_build_object('Authorization', 'Bearer sk_test_XXXXXX')
);

END;

Could you please assist me in ensuring that this query is correct and that it retrieves data from the API as expected? I am unsure if I have entered all the necessary parameters and options correctly.

Thank you in advance for your assistance.

**Comment by GaryAustin1** [comment, marked as answer]
I am assuming you are using the http extension and have it installed.  Note there is also a pg_net extension with the same call but 
http_get does not take headers. https://github.com/pramsey/pgsql-http#functions
You need to use http().
From the same repository link:
Example (You can do the select part like you show I believe):

```
SELECT content::json->'headers'->>'Authorization'
  FROM http((
          'GET',
           'http://httpbun.org/headers',
           ARRAY[http_header('Authorization','Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9')],
           NULL,
           NULL
        )::http_request);
```
        Notice the headers are an array and use http_header and the entire thing is cast to http_request in the call.
        
        You may also need to add `set search_path = public,extensions` in your function definition.

**Reply by GaryAustin1** [reply]
Also you seem to be passing in some values in the 2nd line of your call.  Not sure what those are for a GET.

**Reply by fynn23** [reply]
I want to make an API call to Stripe using a function from Supabase that I can then Call via SQL. I have also activated the extension in Supabase (http). Unfortunately, I am not receiving any values from Stripe with the following query. Do you have any idea what the issue could be?

SELECT content::json->'headers'->>'Authorization'
FROM http((
    'GET',
    'https://api.stripe.com/v1/invoices/in_1OLN6MAXXXXX',
    ARRAY[http_header('Authorization', 'Bearer sk_test_XXXXXX')],
    NULL,
    NULL
)::http_request);

**Reply by GaryAustin1** [reply]
As I said that was an example.  You don't want to return 'headers' as that was the example returned data.
Your original `SELECT status, content::jsonb` looks OK to me without testing.

Also you should look in the dashboard postgres logs for errors.

Also you showed some extra '{"customer": "cus_XXXXXX"}', before.  Is that critical to get to Stripe? If so are you supposed to use GET or POST?

**Reply by fynn23** [reply]
Thank's a lot! My final solution...:

SUPABASE FUNCTION:

DECLARE

invoices jsonb;

BEGIN
    
    SELECT 
        invoice_data
        into invoices
    FROM (
        SELECT 
            jsonb_array_elements(content::jsonb -> 'data') AS invoice_data
        FROM http((
            'GET',
            'https://api.stripe.com/v1/invoices',
            ARRAY[http_header('Authorization', 'Bearer sk_XXXXXXXXXX')],
            NULL,
            NULL
        )::http_request)
    ) AS subquery
    WHERE (invoice_data->>'customer') = customer;

return invoices;

END;
